### PR TITLE
CAF-3262: Document Worker Unit testing - composite documents

### PR DESCRIPTION
**Pull request by kploch**
https://jira.autonomy.com/browse/CAF-3262

Updated the Document Worker Framework unit testing module to support composite documents.

### CODE REVIEW

File: worker-document-testing-unit/src/test/java/com/hpe/caf/worker/document/testing/TestDocumentWorker.java [hunk 1]
```diff
@@ -19,7 +19,13 @@
 import com.hpe.caf.api.worker.DataStoreException;
 import com.hpe.caf.worker.document.exceptions.DocumentWorkerTransientException;
 import com.hpe.caf.worker.document.extensibility.DocumentWorker;
-import com.hpe.caf.worker.document.model.*;
```
Comment (dermot-hardy) [inline worker-document-testing-unit/src/test/java/com/hpe/caf/worker/document/testing/TestDocumentWorker.java:22]: All of the guidelines (Google, Sun, Checkstyle) say to avoid star imports:
http://checkstyle.sourceforge.net/config_imports.html#AvoidStarImport

File: worker-document-testing-unit/src/test/java/com/hpe/caf/worker/document/testing/TestDocumentWorker.java [hunk 2]
```diff
@@ -85,5 +91,12 @@ public void processDocument(Document document) throws InterruptedException, Docu
         for (FieldValue fieldValue : fieldValues) {
             field.add(fieldValue.getStringValue());
         }
+        if (document.hasSubdocuments()) {
```
Comment (dermot-hardy) [inline worker-document-testing-unit/src/test/java/com/hpe/caf/worker/document/testing/TestDocumentWorker.java]: The `if` check seems a bit irrelevant since `.size()` will come back as 0.

**Review by dermot-hardy — APPROVED** [review]

File: worker-document-testing-unit/src/test/java/com/hpe/caf/worker/document/testing/TestDocumentWorker.java (hunk 2 above)
Comment (kploch) [reply worker-document-testing-unit/src/test/java/com/hpe/caf/worker/document/testing/TestDocumentWorker.java]: Yep, I looked into the implementation.

File: worker-document-testing-unit/src/test/java/com/hpe/caf/worker/document/testing/TestDocumentWorker.java (hunk 1 above)
Comment (kploch) [reply worker-document-testing-unit/src/test/java/com/hpe/caf/worker/document/testing/TestDocumentWorker.java:22]: My change removed the star imports. This is because I modified my intellij settings.
